batch/src/main/scala/uk/gov/ons/addressindex/models/HybridAddressNisraEsDocument.scala

```diff
@@ -172,38 +175,84 @@ object HybridAddressNisraEsDocument extends EsDocument {
       "mixedNisra" -> nisraFormatted(0),
       "mixedAltNisra" -> nisraFormatted(1),
       "nisraAll" -> nisraFormatted(2),
-      "organisationName" -> splitAndCapitalise(Option(row.getString(1)).getOrElse("")),
-      "subBuildingName" -> splitAndCapitalise(Option(row.getString(2)).getOrElse("")),
-      "buildingName" -> splitAndCapitalise(Option(row.getString(3)).getOrElse("")),
-      "thoroughfare" -> splitAndCapitalise(Option(row.getString(5)).getOrElse("")),
-      "altThoroughfare" -> splitAndCapitalise(Option(row.getString(6)).getOrElse("")),
-      "dependentThoroughfare" -> splitAndCapitalise(Option(row.getString(7)).getOrElse("")),
-      "locality" -> splitAndCapitalise(Option(row.getString(8)).getOrElse("")),
-      "townland" -> splitAndCapitalise(Option(row.getString(9)).getOrElse("")),
-      "townName" -> splitAndCapitalise(Option(row.getString(10)).getOrElse("")),
-      "postcode" -> row.getString(11)
+      "organisationName" -> splitAndCapitalise(Option(row.getString(15)).getOrElse("")),
+      "subBuildingName" -> splitAndCapitalise(Option(row.getString(1)).getOrElse("")),
+      "buildingName" -> (splitAndCapitalise(Option(row.getString(2)).getOrElse("")) + buildingNameExtra(Option(row.getString(3)).getOrElse("1"))),
+      "thoroughfare" -> splitAndCapitalise(Option(row.getString(16)).getOrElse("")),
+      "altThoroughfare" -> splitAndCapitalise(Option(row.getString(17)).getOrElse("")),
+      "dependentThoroughfare" -> splitAndCapitalise(Option(row.getString(18)).getOrElse("")),
+      "locality" -> splitAndCapitalise(Option(row.getString(19)).getOrElse("")),
+      "townland" -> splitAndCapitalise(Option(row.getString(20)).getOrElse("")),
+      "townName" -> splitAndCapitalise(Option(row.getString(21)).getOrElse("")),
+      "postcode" -> row.getString(22),
+      "complete" -> row.getString(14),
+      "paoText" -> splitAndCapitalise(Option(row.getString(8)).getOrElse("")),
+      "paoStartNumber" -> (if (row.isNullAt(4) || row.getString(4).equals("")) null else toShort(row.getString(4)).getOrElse(null)),
+      "paoStartSuffix" -> row.getString(6),
+      "paoEndNumber" -> (if (row.isNullAt(5) || row.getString(5).equals("")) null else toShort(row.getString(5)).getOrElse(null)),
+      "paoEndSuffix" -> row.getString(7),
+      "saoText" -> splitAndCapitalise(Option(row.getString(13)).getOrElse("")),
+      "saoStartNumber" -> (if (row.isNullAt(9) || row.getString(9).equals("")) null else toShort(row.getString(9)).getOrElse(null)),
+      "saoStartSuffix" -> row.getString(11),
+      "saoEndNumber" -> (if (row.isNullAt(10) || row.getString(10).equals("")) null else toShort(row.getString(10)).getOrElse(null)),
+      "saoEndSuffix" -> row.getString(12)
     )
   }
 
+  def convertToDate(dateTime: String): Timestamp = {
+  //  val formatter = new SimpleDateFormat("yyyy-mm-dd hh:mm:ss")
+    val formatter = new SimpleDateFormat("MM/dd/yy")
+    val utilDate = formatter.parse(dateTime)
```

Comment:
How about: `formatter.parse(dateTime).getTime`. The added `new` smells like Java?

Reply:
This function is not actually being used any more so I'll remove it.